proposals/0075-HID-Support-Plug-in.md

```diff
@@ -149,38 +141,65 @@ Plugin architecture is replaced with a more defined RPC interface. This RPC woul
   </param>
 </struct>
 
-<function name="SendHapticData" functionID="SendHapticDataID" messagetype="request">
+</interface>
 
-  <description>Send the spatial data gathered from SDLCarWindow to the HMI. This data will be utilized by the HMI to determine how and when haptic events should occur</description>
 
-    <param name="HapticSpatialData" type="SpatialStruct" minsize="0" maxsize="50" mandatory="true", array="true">
+<interface name="UI" version="x.x.x" date="yyyy-mm-dd">
+
+<function name="SendHapticData" messagetype="request">
+  <description>Send the UI spatial data from MOBILE. This data will be utilized by the HMI to determine how and when haptic events should occur</description>
+    <param name="appID" type="Integer" mandatory="true">
+      <description>Id of application related to this RPC.</description>
+    </param>
+    <param name="HapticSpatialData" type="Common.SpatialStruct" minsize="0" maxsize="100" mandatory="false", array="true">
       <description>
-        Array of spatial data structures that represent the locations of all user controls present on the HMI. This data should be updated if/when the application presents a new screen.
+        Array of spatial data structures that represent the locations of all user controls present on the HMI. 
+        This data should be updated if/when the application presents a new screen.
+        When a request is sent, if successful, it will replace all spatial data previously sent through RPC.
+        Avoidance of doubt, when an empty HapticSpatialData, it will be clear all spatial data previously sent through RPC.
       </description>
     </param>
 </function>
 
-
-<function name="SendHapticData" functionID="SendHapticDataID" messagetype="response">
-
-    <param name="success" type="Boolean" platform="documentation">
+<function name="SendHapticData" messagetype="response">
+    <param name="success" type="Boolean">
       <description> true if successful; false if failed </description>
     </param>
-
-    <param name="resultCode" type="Result" platform="documentation">
-      <description>See Result</description>
-      <element name="SUCCESS"/>
-      <element name="GENERIC_ERROR"/>
-    </param>
 </function>
 
+</interface>
+
 ```
-This RPC could be adjusted to work in a similar fashion to UI.Show, where each spatial struct has an id, and the proxy only reports deltas when the screen is updated.
+
+
+### New additions to Proxies
```

Comment:
We should call out that a separate proposal describing a more detailed proxy implementation will be entered at a later date.